src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -582,6 +602,15 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompl
             }
         }
 
+        private bool TimeoutFired(CancellationToken callerToken, long timeoutTime) => !callerToken.IsCancellationRequested && timeoutTime > 0 && Environment.TickCount64 >= timeoutTime;
+
+        private static void ThrowTimeoutException(OperationCanceledException originalException)
+        {
+            var timeoutException = new TimeoutException(SR.net_http_request_timedout, originalException);
+            var newTaskCancelledException = new TaskCanceledException(originalException.Message, timeoutException, originalException.CancellationToken);
+            throw newTaskCancelledException;
```

Comment:
I think the message of the top-level exception should be a message about it being canceled due to timeout, e.g.
```C#
throw new TaskCanceledException(
    SR.Format(SR.net_http_request_timedout, Timeout.TotalSeconds),
    new TimeoutException(originalException.Message, originalException));
```
or something like that.

Reply:
Done